pkg/scheduler/scheduler.go

```diff
@@ -419,10 +419,21 @@ func initPolicyFromConfigMap(client clientset.Interface, policyRef *schedulerapi
 
 // Run begins watching and scheduling. It waits for cache to be synced, then starts scheduling and blocked until the context is done.
 func (sched *Scheduler) Run(ctx context.Context) {
+	if metrics.CurrentHostName != "" {
+		metrics.HasLeader.With(map[string]string{"hostname": metrics.CurrentHostName}).Set(1)
```

Comment:
If leader election fails, scheduler won't even run, correct? So I don't think you need `HasLeader=1`, because as long as you have other scheduler metrics such as the `ReadyTime`, it means this scheduler is elected and therefore `HasLeader=1`.

What about using `LeaderElectionFailureCount`?

Reply:
But  it is not easy to use other metrics such as `ReadyTime` to draw out the change of leader.

Reply:
If change of leader is the interested signal, what about having a `leader_change_counter`?